async-query-core/src/main/java/org/opensearch/sql/spark/client/EmrServerlessClientImpl.java

```diff
@@ -116,11 +116,16 @@ public CancelJobRunResult cancelJobRun(
                   } catch (Throwable t) {
                     if (allowExceptionPropagation) {
                       throw t;
-                    } else {
-                      logger.error("Error while making cancel job request to emr:", t);
-                      metricsService.incrementNumericalMetric(EMR_CANCEL_JOB_REQUEST_FAILURE_COUNT);
-                      throw new RuntimeException(GENERIC_INTERNAL_SERVER_ERROR_MESSAGE);
                     }
+
+                    logger.error("Error while making cancel job request to emr:", t);
+                    metricsService.incrementNumericalMetric(EMR_CANCEL_JOB_REQUEST_FAILURE_COUNT);
+                    if (t instanceof ValidationException) {
```

Comment:
Just curious, n which cases will async query core cause EMR to return a validation exception when cancelJob?

Reply:
It is caused when the Job is not cancellable state such as FAILED or COMPLETED. Updated the PR description.

Reply:
What about [ResourceNotFoundException](https://docs.aws.amazon.com/emr-serverless/latest/APIReference/API_CancelJobRun.html#API_CancelJobRun_Errors)

Reply:
That should not happen because we retrieve the job info from AsyncQueryJobMetadata before executing cancel.

Reply:
Is this similar to create job method?

Reply:
Yes, I followed the same way as startJobRun method.